### ISSUE TITLE
Feature request: bypass Wait X after movie is found when movie is add by api call

### ISSUE THREAD

**Issue by SuperPotatoMen** [nosmokingbandit/watcher#217]
I use Ombi (duhh ;) ) to add movies to watcher.
I have one small problem: in couchpotato when you add a movie with a api call, it wil search automatic and snatch when the ETA is past even you have set a delay for waiting X day's.

Is it possible to implentate a option when a movie is added and the eta is past watcher will search and snatch instead of waiting X after movie is found?

**Comment by nosmokingbandit** [comment]
This would be easy enough to do.

Do you think it would be it better to have a global setting in Watcher, or an additional parameter in the api GET request?

`localhost:9090/api?apikey=thisisnotmyapikey&mode=addmovie&imdbid=tt98765&searchnow=true`

**Comment by SuperPotatoMen** [comment]
I like to have a global setting beceause if you add something manually you have the same little problem  :)

**Comment by nosmokingbandit** [comment]
I read through the issue again and misunderstood exactly what you were asking.

I don't use any ETAs to affect search frequency. I haven't found a reliable source of release info (other than physical releases, which isn't super helpful), so I rely on PreDB to tell me when releases have hit the scene.

So I'm trying to figure out the flow for this.

```
                                         If Search Immediately enabled:  If found and Automatically Grab Result enabled and Days to Wait is 0:
User adds movie > Watcher checks predb > Watcher searches indexers     > Watcher grabs result

```

So you want to be able to skip the Wait X Days on what what condition?

**Comment by SuperPotatoMen** [comment]
@nosmokingbandit 
I was wondering what is wrong with the physical releases dates? For example, the best releases are coming from the physical release. Maybe you can make it optional that the enduser can choose between different search & snatch delays? Something like: Theater release, physical release and PreDB.

**Comment by nosmokingbandit** [comment]
Generally, if something doesn't exist on PreDB it isn't worth downloading.
I'd rather miss a release for a few days than download/process/mark a fake
release.
If someone wants a WEBDL it makes no sense to wait for the physical release
date since they'll likely be able to get it much earlier. A certain
anti-hero movie was available digitally about a month before physical
copies were available.

**Comment by SuperPotatoMen** [comment]
@nosmokingbandit that is correct, thats why i'm suggest to make it optional for movie lovers for good quality :)

But is it possible when a movie is added and it is on PreDB that you have a option to skip the wait X days for better quality? The reason why i'm asking this is beceause if a user request something on Ombi and it is released 3 years ago, it will be nice if watcher skip the X waiting days and start searching and snatching beceause good quality releases are already on newsgroup. Otherway the requester need to wait 3 or 5 days until he can watch it :(

**Comment by SuperPotatoMen** [comment]
@nosmokingbandit that is correct, thats why i'm suggest to make it optional for movie lovers for good quality :)

But is it possible when a movie is added and it is on PreDB that you have a option to skip the wait X days for better quality? The reason why i'm asking this is beceause if a user request something on Ombi and it is released 3 years ago, it will be nice if watcher skip the X waiting days and start searching and snatching beceause good quality releases are already on newsgroup. Otherway the requester need to wait 3 or 5 days until he can watch it :( 

So basically: the only thing i'm missing in watcher is a option to skip the waiting days when the movie is already on PreDB.

**Comment by nosmokingbandit** [comment]
This makes sense. I'm trying to think of a good way to execute this without adding too much bulk to the search process.

**Comment by SuperPotatoMen** [comment]
@nosmokingbandit can i make a suggestion?
I'm not sure but to keep it simple: if it is possible parse the date on PreDB, is it behind a month in the past? ---> Skip the delay function, not  behind a month in the past? activate the delay function.
You can turn this on/off of in the setting page.

**Comment by nosmokingbandit** [comment]
I finally made it back to this.

The thing about your suggestion is that the [predb search](https://predb.me/?search=rogue+one&rss=1) doesn't give dates unless I parse the html. I don't like parsing html because minor changes on their side can break the whole thing and something as critical as predb checking shouldn't be able to be broken by adding a `<br/>` to the html.

Commit 19927b29c6e705e4c37c8df92634ccfc4a1ce8a5 is the best I could do without adding tons of extra tmdb api calls or parse html.

Basically it checks the theatrical release date and counts weeks from that. So I could set it to 15 weeks. I add Rogue One and it won't download immediately. I add Police Academy and it grabs the torrent immediately.

Its not ideal, but it gets the job done.

**Comment by SuperPotatoMen** [comment]
@nosmokingbandit  wow thanks dude for the amezing worl
